### PR TITLE
getArcGISCSVURL without the server name

### PR DESCRIPTION
**Pull request by lazd**
Gets the full manifest and finds the data source based on
widget name.  From the data source, you can get the server.

** This code has not been run.  Please validate if you merge it **

Supersedes #382

### CODE REVIEW

File: src/shared/lib/fetch/index.js [hunk 1]
```diff
@@ -225,3 +225,66 @@ export const getArcGISCSVURL = async function(serverNumber, dashboardId, layerNa
   const { orgId } = dashboardManifest;
   return getArcGISCSVURLFromOrgId(serverNumber, orgId, layerName);
 };
+
+/**
+ * Find data source id for a widget from the manifest
+ *
+ * @param {*} dashboardId the ID of the dashboard
+ * @param {*} widgetName the name of the widget in the manifest
```
Comment (camjc) [inline src/shared/lib/fetch/index.js]:
```suggestion
 * @param {string} dashboardId the ID of the dashboard
 * @param {string} widgetName the name of the widget in the manifest
```

File: src/shared/lib/fetch/index.js [hunk 2]
```diff
@@ -225,3 +225,66 @@ export const getArcGISCSVURL = async function(serverNumber, dashboardId, layerNa
...
+      return w.datasets[0].dataSource.itemId;
+    }
+  }
+  throw new Error('Cannot find widget');
+};
+
+/**
+ * Get the URL for the CSV data from an ArcGIS dashboard
+ * @param {*} dashboardId the ID of the dashboard
+ * @param {*} widgetName the name of the widget in the manifest
```
Comment (camjc) [inline src/shared/lib/fetch/index.js]:
```suggestion
 * @param {string} dashboardId the ID of the dashboard
 * @param {string} widgetName the name of the widget in the manifest
```

File: src/shared/lib/fetch/index.js [hunk 3]
```diff
@@ -225,3 +225,66 @@ export const getArcGISCSVURL = async function(serverNumber, dashboardId, layerNa
...
+  // 3. get the orgID, layerName, and url from the data source
+  const sourceInfo = await json(`https://maps.arcgis.com/sharing/rest/content/items/${datasourceId}?f=json`);
+  const xorgId = sourceInfo.orgId;
+  const layerName = sourceInfo.name;
+  const serverName = sourceInfo.url.substr(0, 28);
+
+  // error checks...
+  if (!xorgId) throw new Error('Missing orgID');
+  if (!layerName) throw new Error('Missing layerName');
+  if (!serverName.endsWith('.com')) throw new Error('Could not parse server name');
```
Comment (camjc) [inline src/shared/lib/fetch/index.js]: Would need to add the import too, semantically feels like assertions, so what about
```suggestion
assert(xorgId, "Missing orgID");
assert(layerName, "Missing layerName");
assert(serverName.endsWith(".com"), "Could not parse server name");
```

**Comment by lazd** [comment]
This has gone stale, and it's not used by any scrapers. Closing.

**Pull request closed without merging**